All CI runs of `pytest-minimal-install` are suddenly failing

All CI runs started failing this afternoon in the `pytest-minimal-install` job. Below is an example. The most proximal cause that it's now installing Cirq 1.7.0, which was uploaded to PyPI this afternoon.

<details>
  <summary><code> FAILURES </code></summary>
<pre>
=================================== FAILURES ===================================
__________________________________ test_gates __________________________________

    def test_gates():
        # T gate and its adjoint
        assert _from_directly_countable_cirq(cirq.T) == TComplexity(t=1)
        assert _from_directly_countable_cirq(cirq.T**-1) == TComplexity(t=1)
    
        assert _from_directly_countable_cirq(cirq.H) == TComplexity(clifford=1)  # Hadamard
        assert _from_directly_countable_cirq(cirq.CNOT) == TComplexity(clifford=1)  # CNOT
        assert _from_directly_countable_cirq(cirq.S) == TComplexity(clifford=1)  # S
        assert _from_directly_countable_cirq(cirq.S**-1) == TComplexity(clifford=1)  # S†
    
        # Pauli operators are clifford
        assert _from_directly_countable_cirq(cirq.X) == TComplexity(clifford=1)
        assert _from_directly_countable_cirq(cirq.Y) == TComplexity(clifford=1)
        assert _from_directly_countable_cirq(cirq.Z) == TComplexity(clifford=1)
    
        # Rotation about X, Y, and Z axes
        assert _from_directly_countable_cirq(cirq.Rx(rads=2)) == TComplexity(rotations=1)
        assert _from_directly_countable_cirq(cirq.Ry(rads=2)) == TComplexity(rotations=1)
        assert _from_directly_countable_cirq(cirq.Rz(rads=2)) == TComplexity(rotations=1)
    
        # clifford+T
        assert t_complexity(And()) == TComplexity(t=4, clifford=9)
        assert t_complexity(And() ** -1) == TComplexity(clifford=4)
    
        assert t_complexity_compat(cirq.FREDKIN) == TComplexity(t=7, clifford=14)
    
        # TODO: https://github.com/quantumlib/Qualtran/issues/237
>       assert t_complexity_compat(cirq.H.controlled()) == TComplexity(
            clifford=4, rotations=2
        ) or t_complexity_compat(  # cirq <=1.4
            cirq.H.controlled()
        ) == TComplexity(
            t=2, clifford=4, rotations=2
        )
E       AssertionError: assert (TComplexity(t..., rotations=1) == TComplexity(t..., rotations=2)
E         
E         Differing attributes:
E         ['t', 'clifford', 'rotations']
E         
E         Drill down into differing attribute t:
E           t: 1 != 0
E         
E         Drill down into differing attribute clifford:
E           clifford: 3 != 4
E         
E         Drill down into differing attribute rotations:
E           rotations: 1 != 2 or TComplexity(t..., rotations=1) == TComplexity(t..., rotations=2)
E         
E         Differing attributes:
E         ['t', 'clifford', 'rotations']
E         
E         Drill down into differing attribute t:
E           t: 1 != 2
E         
E         Drill down into differing attribute clifford:
E           clifford: 3 != 4
E         
E         Drill down into differing attribute rotations:
E           rotations: 1 != 2)

qualtran/cirq_interop/t_complexity_protocol_test.py:123: AssertionError
________________________ test_select_hubbard_roundtrip _________________________

    def test_select_hubbard_roundtrip():
>       check_roundtrip(
            'qualtran.bloqs.chemistry.hubbard_model.qubitization.SelectHubbard(x_dim=5, y_dim=5)',
            qualtran.bloqs.chemistry.hubbard_model.qubitization.SelectHubbard(5, 5),
        )

qualtran/l1/_roudtrip_test.py:145: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

bloq_objectstring = 'qualtran.bloqs.chemistry.hubbard_model.qubitization.SelectHubbard(x_dim=5, y_dim=5)'
bloq = SelectHubbard(x_dim=5, y_dim=5, control_val=None)

    def check_roundtrip(bloq_objectstring: str, bloq: qlt.Bloq):
    
        loaded_bloq = qualtran.l1.load_bloq(bloq_objectstring)
        assert loaded_bloq == bloq
    
        # Manually use L1ModuleBuilder so we have a record of all the true subbloqs
        # we're serializing.
        l1mb = qualtran.l1.L1ModuleBuilder()
        _root_bloq_key = l1mb.add_bloqs(root=bloq, extern_only_from=False)
        original_bloqs = {qdef.qdef.bloq_key: qdef for qdef in l1mb.qdefs}
    
        # Get the textual representation
        l1_code = qualtran.l1.L1ASTPrinter().visit(l1mb.finalize())
    
        # Re-parse the textual representation
        l1_module = qualtran.l1.load_module(l1_code)
    
        problems = []
        problems += check_bloq_keys(original_bloqs, l1_module)
        problems += check_signatures(original_bloqs, l1_module)
        problems += check_bloq_object_roundtrip(original_bloqs, l1_module)
        problems += check_soquet_graph_isomorphism(original_bloqs, l1_module)
>       assert problems == []
E       AssertionError: assert ['BOBJ: FAIL cirq.CY'] == []
E         
E         Left contains one more item: 'BOBJ: FAIL cirq.CY'
E         
E         Full diff:
E         - []
E         + [
E         +     'BOBJ: FAIL cirq.CY',
E         + ]

qualtran/l1/_roudtrip_test.py:122: AssertionError
------------------------------ Captured log call -------------------------------
WARNING  qualtran.l1._to_l1:_to_l1.py:509 Structured __str__ for Split found, but had to be disambiguated, i=1
WARNING  qualtran.l1._to_l1:_to_l1.py:509 Structured __str__ for Join found, but had to be disambiguated, i=1
WARNING  qualtran.l1._to_l1:_to_l1.py:509 Structured __str__ for Split found, but had to be disambiguated, i=2
WARNING  qualtran.l1._to_l1:_to_l1.py:509 Structured __str__ for Join found, but had to be disambiguated, i=2
WARNING  qualtran.l1._to_l1:_to_l1.py:509 Structured __str__ for Split found, but had to be disambiguated, i=3
WARNING  qualtran.l1._to_l1:_to_l1.py:509 Structured __str__ for Join found, but had to be disambiguated, i=3
ERROR    qualtran.l1._eval:_eval.py:448 Failed to load the corresponding bloq 'from' for cirq.CY: Tried to evaluate an AST node that was not properly serialized: CObjectNode(name='Unserializable', cargs=(CArgNode(key=None, value=LiteralNode(value='CYPowGate')),))
</pre>
</details>

--- a/qualtran/cirq_interop/_cirq_to_bloq.py
+++ b/qualtran/cirq_interop/_cirq_to_bloq.py
@@ -416,14 +416,15 @@ def cirq_gate_to_bloq(gate: cirq.Gate) -> Bloq:
         cirq.TOFFOLI: Toffoli(),
         cirq.X: XGate(),
         cirq.Y: YGate(),
-        # cirq.CY in Cirq >= 1.7
-        getattr(cirq, 'CY', cirq.ControlledGate(cirq.Y)): CYGate(),
+        cirq.ControlledGate(cirq.Y): CYGate(),
         cirq.Z: ZGate(),
         cirq.CZ: CZ(),
         cirq.SWAP: TwoBitSwap(),
         cirq.CSWAP: CSwap(1),
         cirq.I: Identity(),
     }
+    if hasattr(cirq, 'CY'):
+        CIRQ_GATE_TO_BLOQ_MAP[cirq.CY] = CYGate()  # cirq >= 1.7
     if gate in CIRQ_GATE_TO_BLOQ_MAP:
         return CIRQ_GATE_TO_BLOQ_MAP[gate]
 

--- a/qualtran/cirq_interop/_cirq_to_bloq.py
+++ b/qualtran/cirq_interop/_cirq_to_bloq.py
@@ -416,7 +416,7 @@ def cirq_gate_to_bloq(gate: cirq.Gate) -> Bloq:
         cirq.TOFFOLI: Toffoli(),
         cirq.X: XGate(),
         cirq.Y: YGate(),
-        cirq.ControlledGate(cirq.Y): CYGate(),
+        getattr(cirq, 'CY', cirq.ControlledGate(cirq.Y)): CYGate(),
         cirq.Z: ZGate(),
         cirq.CZ: CZ(),
         cirq.SWAP: TwoBitSwap(),

--- a/qualtran/cirq_interop/_cirq_to_bloq.py
+++ b/qualtran/cirq_interop/_cirq_to_bloq.py
@@ -416,6 +416,7 @@ def cirq_gate_to_bloq(gate: cirq.Gate) -> Bloq:
         cirq.TOFFOLI: Toffoli(),
         cirq.X: XGate(),
         cirq.Y: YGate(),
+        # cirq.CY in Cirq >= 1.7
         getattr(cirq, 'CY', cirq.ControlledGate(cirq.Y)): CYGate(),
         cirq.Z: ZGate(),
         cirq.CZ: CZ(),

--- a/qualtran/cirq_interop/t_complexity_protocol.py
+++ b/qualtran/cirq_interop/t_complexity_protocol.py
@@ -25,6 +25,7 @@ from .decompose_protocol import _decompose_once_considering_known_decomposition
 _T_GATESET = cirq.Gateset(cirq.T, cirq.T**-1, unroll_circuit_op=False)
 _ROTS_GATES = [cirq.XPowGate, cirq.YPowGate, cirq.ZPowGate, cirq.CZPowGate]
 if hasattr(cirq, 'CYPowGate'):
+    # cirq.CY in Cirq >= 1.7
     _ROTS_GATES.append(cirq.CYPowGate)
 _ROTS_GATESET = cirq.Gateset(*_ROTS_GATES)
 
@@ -168,7 +169,7 @@ def _t_complexity_from_strategies(
 
 @cachetools.cached(cachetools.LRUCache(128), key=_get_hash, info=True)
 def _t_complexity_for_gate_or_op(
-    gate_or_op: Union[cirq.Gate, cirq.Operation, Bloq]
+    gate_or_op: Union[cirq.Gate, cirq.Operation, Bloq],
 ) -> Optional[TComplexity]:
     if isinstance(gate_or_op, cirq.Operation) and gate_or_op.gate is not None:
         gate_or_op = gate_or_op.gate

--- a/qualtran/cirq_interop/t_complexity_protocol.py
+++ b/qualtran/cirq_interop/t_complexity_protocol.py
@@ -23,7 +23,10 @@ from qualtran.symbolics import ceil, log2, SymbolicFloat, SymbolicInt
 from .decompose_protocol import _decompose_once_considering_known_decomposition
 
 _T_GATESET = cirq.Gateset(cirq.T, cirq.T**-1, unroll_circuit_op=False)
-_ROTS_GATESET = cirq.Gateset(cirq.XPowGate, cirq.YPowGate, cirq.ZPowGate, cirq.CZPowGate)
+_ROTS_GATES = [cirq.XPowGate, cirq.YPowGate, cirq.ZPowGate, cirq.CZPowGate]
+if hasattr(cirq, 'CYPowGate'):
+    _ROTS_GATES.append(cirq.CYPowGate)
+_ROTS_GATESET = cirq.Gateset(*_ROTS_GATES)
 
 
 @attrs.frozen

--- a/qualtran/cirq_interop/t_complexity_protocol.py
+++ b/qualtran/cirq_interop/t_complexity_protocol.py
@@ -23,7 +23,7 @@ from qualtran.symbolics import ceil, log2, SymbolicFloat, SymbolicInt
 from .decompose_protocol import _decompose_once_considering_known_decomposition
 
 _T_GATESET = cirq.Gateset(cirq.T, cirq.T**-1, unroll_circuit_op=False)
-_ROTS_GATES = [cirq.XPowGate, cirq.YPowGate, cirq.ZPowGate, cirq.CZPowGate]
+_ROTS_GATES: list[type[cirq.Gate]] = [cirq.XPowGate, cirq.YPowGate, cirq.ZPowGate, cirq.CZPowGate]
 if hasattr(cirq, 'CYPowGate'):
     # cirq.CY in Cirq >= 1.7
     _ROTS_GATES.append(cirq.CYPowGate)

--- a/qualtran/cirq_interop/t_complexity_protocol_test.py
+++ b/qualtran/cirq_interop/t_complexity_protocol_test.py
@@ -120,13 +120,11 @@ def test_gates():
     assert t_complexity_compat(cirq.FREDKIN) == TComplexity(t=7, clifford=14)
 
     # TODO: https://github.com/quantumlib/Qualtran/issues/237
-    assert t_complexity_compat(cirq.H.controlled()) == TComplexity(
-        clifford=4, rotations=2
-    ) or t_complexity_compat(  # cirq <=1.4
-        cirq.H.controlled()
-    ) == TComplexity(
-        t=2, clifford=4, rotations=2
-    )
+    assert t_complexity_compat(cirq.H.controlled()) in [
+        TComplexity(clifford=4, rotations=2),  # cirq >= 1.5
+        TComplexity(t=2, clifford=4, rotations=2),  # cirq <= 1.4
+        TComplexity(t=1, clifford=1, rotations=1),  # cirq >= 1.7 (CYPowGate)
+    ]
     assert t_complexity(CHadamard()) == TComplexity(clifford=4, rotations=2)
 
     # Global phase

